src/components/icon/icon.ts

```diff
@@ -87,46 +85,62 @@ export default class SlIcon extends ShoelaceElement {
     const library = getIconLibrary(this.library);
     const url = this.getUrl();
 
-    // Create an instance of the DOM parser. We do it here instead of top-level to support SSR while maintaining a
```

Comment:
I'm not really clear on why this is lazy initializing the DOMParser. Should I be doing this for iconCache too?

Reply:
IIRC this change was made working with the Lit team to [support SSR](#778) and we realized that `DOMParser` isn't an API that's available on the backend. Instantiation was originally top-level, which caused SSR to error out. Moving it here solved for that.

Reply:
Maps should be fine.